src/App.jsx

```diff
@@ -1,18 +1,28 @@
 // src/App.jsx
 import { useState } from 'react';
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
 import './App.css';
 import Login from './components/Login';
 import Signup from './components/Signup';
+import MainPage from './pages/MainPage';
+import ConcertDetailsPage from './pages/ConcertDetailsPage';
 
 function App() {
   const [showLogin, setShowLogin] = useState(true);
 
   return (
-    <div className='App'>
-      <button onClick={() => setShowLogin(true)}>Login</button>
-      <button onClick={() => setShowLogin(false)}>Signup</button>
-      {showLogin ? <Login /> : <Signup />}
-    </div>
+    <Router>
+      <div className='App'>
+        <button onClick={() => setShowLogin(true)}>Login</button>
+        <button onClick={() => setShowLogin(false)}>Signup</button>
+        {showLogin ? <Login /> : <Signup />}
```

Comment:
I think would be better if this is included within its own component, I think would make sense to include this in your authentication component, and then import it and render it only if user is not authenticated, other wise redirect to home page. Keeping components within their own file would reduce conflicts and will make easire future modifications!